dts/bindings/net/wireless/nordic,nrf-radio.yaml

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Nordic DFE antennas matrix description
+
+    dfegpio[n]-gpios properties provide GPIO pins to Radio peripheral. Provided
+    GPIO pins are used to drive antennas switching by Radio Direction Finding
+    Extension. E.g. to use 12 antennas 4 GPIO pins must be provided,
+    4 GPIO pins support switching up to 16 antennas.
+
+    All dfegpio[n]-gpios properties have required property set to false.
+    It depends on end user which Radio PSEL_GPIO[n] registers will be used
+    to drive antenna switches.
+    Nevertheless there is a requirement that at least two antennas are
+    available. That means at least one dfegpio[n]-gpios property must be
+    provided, to give Radio possibility to switch between these two antennas.
+
+compatible: "nordic,nrf-radio"
+
+include: [base.yaml]
+
+properties:
+    ant-num:
+      type: int
+      required: true
+      description: |
+        Number of available antennas.
+
+        Value zero means no antenna available. If provided value different than
+        zero, it must be greater or equal to two.
+
+    dfegpio0-gpios:
```

Comment:
I think somewhere there was discussion about whether things like this should be instead `dfegpio-gpios`, with the value being a sequence of phandle+specifiers.  IIRC that was at one point prevented by the inability to express an absent entry, but I think that was to be fixed, e.g.:
```
dfe-gpios = <&gpio0 0 0>, 0, <&gpio0 2 0>;
```

I have no position on this, but the array form would eliminate some verbosity.

Reply:
I've decided to go with `dfegpios#-gpios`, because it gives end user possibility to use e.g. GPIO 1, 2, 5, 6.
Also I wanted to avoid use of programming macros in first attempt. It changed during review of @anangl.